File: zenyogaui/packages/core/lib/dto/responses/groupped_classes.dart

import 'package:core/dto/responses/class_response_dto.dart';
import 'package:json_annotation/json_annotation.dart';

part 'groupped_classes.g.dart';
@JsonSerializable()
class GrouppedClasses {
  final List<ClassResponseDto> hathaYoga;
  final List<ClassResponseDto> yinYoga;
  final List<ClassResponseDto> vinyasaYoga;


  GrouppedClasses({
    required this.hathaYoga,
    required this.yinYoga,
    required this.vinyasaYoga
  });


  factory GrouppedClasses.fromJson(Map<String, dynamic> json) => _$GrouppedClassesFromJson(json);

  Map<String, dynamic> toJson() => _$GrouppedClassesToJson(this);
}

File: zen_yoga_api/ZEN&Yoga.Models/Responses/GrouppedClasses.cs
namespace ZEN_Yoga.Models.Responses
{
    public class GrouppedClasses
    {
        public List<ClassResponse> HathaYoga { get; set; } = new List<ClassResponse>();

        public List<ClassResponse> YinYoga { get; set; } = new List<ClassResponse>();

        public List<ClassResponse> VinyasaYoga { get; set; } = new List<ClassResponse>();
    }
}
